### PR TITLE
Don't cache in target_history()

### LINKED ISSUE #1956: backend.target_history() caches backend properties
**Describe the bug**

`backend.target_history()` calls `_get_properties()`:

https://github.com/Qiskit/qiskit-ibm-runtime/blob/51d897bee4c7cf28e22b6a0a0c9b2f337c23a9ba/qiskit_ibm_runtime/ibm_backend.py#L349-L355

which saves the properties of that given timestamp as `_properties`:

https://github.com/Qiskit/qiskit-ibm-runtime/blob/51d897bee4c7cf28e22b6a0a0c9b2f337c23a9ba/qiskit_ibm_runtime/ibm_backend.py#L241-L249

which means if I do 

```python
old_history = backend.target_history(datetime=<1 year ago>)
props = backend.properties()
```

I'll actually get `old_history` back in my second call. 

**Steps to reproduce**

**Expected behavior**

Historical properties should not be cached. `backend.properties()` does this correctly. 

**Suggested solutions**

Don't cache properties if datetime is provided. 

**Additional Information**

- **qiskit-ibm-runtime version**:
- **Python version**:
- **Operating system**:

### PR DESCRIPTION
**Pull request by kt474**
### Details and comments
Fixes #1956

### CODE REVIEW

File: qiskit_ibm_runtime/ibm_backend.py [hunk 1]
```diff
@@ -355,10 +345,23 @@ def target_history(self, datetime: Optional[python_datetime] = None) -> Target:
         Returns:
             Target with properties found on `datetime`
         """
-        self._get_properties(datetime=datetime)
         self._get_defaults()
-        self._convert_to_target(refresh=True)
-        return self._target
+        if self.options.use_fractional_gates is None:
+            include_control_flow = True
+            include_fractional_gates = True
```
Comment (ptristan3) [inline qiskit_ibm_runtime/ibm_backend.py:351]: These are the default values for convert_to_target()

**Review by ptristan3 — APPROVED** [review]
LGTM

File: qiskit_ibm_runtime/ibm_backend.py [hunk 2]
```diff
@@ -261,21 +251,14 @@ def _get_defaults(self, refresh: bool = False) -> None:
     def _convert_to_target(self, refresh: bool = False) -> None:
         """Converts backend configuration, properties and defaults to Target object"""
         if refresh or not self._target:
-            if self.options.use_fractional_gates is None:
-                include_control_flow = True
-                include_fractional_gates = True
-            else:
-                # In IBM backend architecture as of today
-                # these features can be only exclusively supported.
```
Comment (jyu00) [inline qiskit_ibm_runtime/ibm_backend.py:269]: Can we preserve these comments?

File: qiskit_ibm_runtime/ibm_backend.py [hunk 3]
```diff
@@ -239,16 +239,6 @@ def __getattr__(self, name: str) -> Any:
...
-        """Gets backend properties and decodes it"""
-        if datetime:
-            datetime = local_to_utc(datetime)
-        if datetime or not self._properties:
-            api_properties = self._api_client.backend_properties(self.name, datetime=datetime)
-            if api_properties:
-                backend_properties = properties_from_server_data(api_properties)
-                self._properties = backend_properties
-
     def _get_defaults(self, refresh: bool = False) -> None:
```
Comment (jyu00) [inline qiskit_ibm_runtime/ibm_backend.py:242]: This can be a separate PR, but I noticed there is a `_get_defaults()` which seems to do the same thing as `defaults()`

**Review by jyu00 — APPROVED** [review]
Some minor comments but otherwise looks good

File: qiskit_ibm_runtime/ibm_backend.py (hunk 3 above)
Comment (kt474) [reply qiskit_ibm_runtime/ibm_backend.py:242]: I'll handle this in a separate PR

Comment (kt474) [reply qiskit_ibm_runtime/ibm_backend.py:242]: We can use #1910
